Sources/SwiftInspectorVisitors/Tests/ClassVisitorSpec.swift

```diff
@@ -1,413 +0,0 @@
-            }
-
-            expect(matching.count) == 1
-          }
-
-          it("finds FooClass.BarFooEnum.BarBarFooClass") {
-            let matching = self.sut.classes.filter {
-              $0.name == "BarBarFooClass"
-                && $0.inheritsFromTypes.map { $0.asSource } == []
-                && $0.parentType?.asSource == "FooClass.BarFooEnum"
-            }
-
-            expect(matching.count) == 1
-          }
-        }
-      }
-
-      context("visiting a code block with multiple top-level declarations") {
-        context("with multiple top-level classes") {
-          it("asserts") {
-            let content = """
-            public class FooClass {}
-            public class BarClass {}
-            """
-
-            // The ClassVisitor is only meant to be used over a single class.
-            // Using a ClassVisitor over a block that has multiple top-level
-            // classes is API misuse.
-            expect(try VisitorExecutor.walkVisitor(
-                    self.sut,
-                    overContent: content))
-              .to(throwAssertion())
-          }
-        }
-
-        context("with a top-level struct after a top-level class") {
-          it("asserts") {
-            let content = """
-            public class FooClass {}
-            public struct FooStruct {}
-            """
-
-            // The ClassVisitor is only meant to be used over a single class.
-            // Using a ClassVisitor over a block that has a top-level struct
-            // is API misuse.
-            expect(try VisitorExecutor.walkVisitor(
-                    self.sut,
-                    overContent: content))
-              .to(throwAssertion())
-          }
-        }
-
-        context("with a top-level enum after a top-level class") {
-          it("asserts") {
-            let content = """
-            public class FooClass {}
-            public struct FooEnum {}
-            """
-
-            // The ClassVisitor is only meant to be used over a single class.
-            // Using a ClassVisitor over a block that has a top-level enum
-            // is API misuse.
-            expect(try VisitorExecutor.walkVisitor(
-                    self.sut,
-                    overContent: content))
-              .to(throwAssertion())
-          }
-        }
-      }
-
-      context("visiting a code block with a top-level struct declaration") {
-        it("asserts") {
-          let content = """
-            public struct FooStruct {}
-            """
-
-          // The ClassVisitor is only meant to be used over a single class.
-          // Using a ClassVisitor over a block that has a top-level struct
-          // is API misuse.
-          expect(try VisitorExecutor.walkVisitor(
-                  self.sut,
-                  overContent: content))
-            .to(throwAssertion())
-        }
-      }
-
-      context("visiting a code block with a top-level enum declaration") {
```

Comment:
this test was not moved since a `NestableDeclarationVisitor` can parse this declaration type